lib/new_relic/agent/instrumentation/grape.rb

```diff
@@ -19,9 +19,16 @@
 
   depends_on do
     begin
-      if defined?(Bundler) && Bundler.rubygems.all_specs.map(&:name).include?('newrelic-grape')
-        NewRelic::Agent.logger.info('Not installing New Relic supported Grape instrumentation because the third party newrelic-grape gem is present')
-        false
+      if defined?(Bundler)
+        grape_present = (Bundler::VERSION >= '2' && Bundler.rubygems.installed_specs.map(&:name).include?('newrelic-grape')) ||
+          Bundler.rubygems.all_specs.map(&:name).include?('newrelic-grape')
+
+        if grape_present
+          NewRelic::Agent.logger.info('Not installing New Relic supported Grape instrumentation because the third party newrelic-grape gem is present')
+          false
+        else
+          true
+        end
```

Comment:
```suggestion
      if (defined?(Bundler) && (Bundler::VERSION >= '2' && Bundler.rubygems.installed_specs.map(&:name).include?('newrelic-grape')) ||
        Bundler.rubygems.all_specs.map(&:name).include?('newrelic-grape'))
        NewRelic::Agent.logger.info('Not installing New Relic supported Grape instrumentation because the third party newrelic-grape gem is present')
        false
```